nm: Change NetworkManager plugin as optional

cathay4t opened the pull request:
Previously, if NetworkManager plugin is not properly loaded, we raise
exception of it. This stop user from using their own backend.

Changed the NetworkManager plugin as optional by only log a warning
indicating plugin is not loaded with reason.

Integration test case included.

ffmancera commented on libnmstate/nm/common.py:
```diff
@@ -19,15 +19,12 @@
-from gi.repository import GLib
-from gi.repository import GObject
-from gi.repository import Gio
+from gi.repository import NM  # NOQA: E402
+from gi.repository import GLib  # NOQA: E402
+from gi.repository import GObject  # NOQA: E402
+from gi.repository import Gio  # NOQA: E402
```
Why is `NOQA: E402` needed? Thanks!

cathay4t replied:
https://www.flake8rules.com/rules/E402.html

cathay4t replied:
Add comment

```
# It is required to state the NM version before importing it
# But this break the flak8 rule: https://www.flake8rules.com/rules/E402.html
# Use NOQA: E402 to suppress it.
```

ffmancera reviewed (approved):
Looks good to me, I've made a comment below. Thank you!

ffmancera reviewed (changes requested):
black is complaining on formatting, please check this out. Thanks!

cathay4t committed:
nm: Change NetworkManager plugin as optional

Previously, if NetworkManager plugin is not properly loaded, we raise
exception of it. This stop user from using their own backend.

Changed the NetworkManager plugin as optional by only log a warning
indicating plugin is not loaded with reason.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
